src/reducers/categories.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Created by Katarzyna_Bak on 18.05.2017.
+ */
+
+const initialState = [
+    {id: 1, name: "Category 1", parentId: null },
+    {id: 2, name: "Category 2", parentId: null },
+    {id: 3, name: "Category 3", parentId: null },
+    {id: 4, name: "Category 4", parentId: null },
+    {id: 5, name: "Category 5", parentId: 1 },
+    {id: 6, name: "Category 6", parentId: 1 },
+    {id: 7, name: "Category 7", parentId: 2 },
+    {id: 8, name: "Category 8", parentId: 4 },
+    {id: 9, name: "Category 9", parentId: 4 },
+    {id: 10, name: "Category 10", parentId: 4 },
+];
+
+let nextCategoryId = 11;
+
+const categories = (state = initialState, action) => {
+    switch(action.type) {
+        case 'ADD_CATEGORY':
+            return [
+                ...state, {
+                    id: nextCategoryId++,
```

Comment:
Reducer must be a pure function, so it shouldn't rely on outer scope's `nextCategoryId`. Either store it in state or calculate `id` basing on categories.